acceptance-tests/src/test/java/tech/pegasys/ethsigner/tests/hashicorpVault/HashicorpVaultDocker.java

```diff
@@ -0,0 +1,222 @@
+    "vault",
+    "kv",
+    "put",
+    "secret/ethsignerSigningKey",
+    "value=8f2a55949038a9610f50fb23b5883af3b4ecb3c3bb792cbcefbd1542c692be63"
+  };
+
+  private final DockerClient docker;
+  private final String vaultContainerId;
+
+  private int port;
+
+  public HashicorpVaultDocker(final DockerClient docker) {
+    this.docker = docker;
+    pullVaultImage();
+    this.vaultContainerId = createVaultContainer();
+  }
+
+  public void start() {
+    LOG.info("Starting Hashicorp Vault Docker container: {}", vaultContainerId);
+    docker.startContainerCmd(vaultContainerId).exec();
+
+    LOG.info("Querying for the Docker dynamically allocated vault port number");
+    final InspectContainerResponse containerResponse =
+        docker.inspectContainerCmd(vaultContainerId).exec();
+    final Ports ports = containerResponse.getNetworkSettings().getPorts();
+    port = httpRpcPort(ports);
+    LOG.info("Http port for Hashicorp Vault: {}", port);
+
+    try {
+      TimeUnit.SECONDS.sleep(10); // wait until the docker is container is all up and running.
+    } catch (InterruptedException e) {
+      LOG.error("Interruption while waiting for Hashicorp Vault to be up");
+    }
+
+    // After starting the docker container with the vault we need to create the secret that contains
+    // the
+    // the private key. That is done in the awaitStartupCompletion, because we need to wait until a
+    // call to
+    // the vault succeeds to know that it is up. That call now creates the secret.
+  }
+
+  private ExecCreateCmdResponse getExecCreateCmdResponse(String[] commandWithArguments) {
+    return docker
+        .execCreateCmd(vaultContainerId)
+        .withAttachStdout(true)
+        .withAttachStderr(true)
+        .withCmd(commandWithArguments)
+        .exec();
+  }
+
+  private boolean runCommandInVaultContainer(
+      ExecCreateCmdResponse execCreateCmdResponse, String expectedInStdout)
+      throws InterruptedException {
+    ByteArrayOutputStream stdout = new ByteArrayOutputStream();
+    ByteArrayOutputStream stderr = new ByteArrayOutputStream();
+    final ExecStartResultCallback resultCallback = new ExecStartResultCallback(stdout, stderr);
+    final ExecStartResultCallback execStartResultCallback =
+        docker.execStartCmd(execCreateCmdResponse.getId()).exec(resultCallback).awaitCompletion();
+    execStartResultCallback.onError(
+        new RuntimeException(
+            "command in Hashicorp Vault returned error\n" + execStartResultCallback.toString()));
+    if (expectedInStdout != null) {
+      final String stdoutString = stdout.toString();
+      return stdoutString.indexOf(expectedInStdout) != -1;
+    } else {
+      return true;
+    }
+  }
+
+  public void shutdown() {
+    if (hasVaultContainer()) {
+      stopVaultContainer();
+      removeVaultContainer();
+    }
+  }
+
+  /** This method now also creates the secret that contains the private signing key. */
+  public void awaitStartupCompletion() {
```

Comment:
seems a bit odd that this method modifies the vault environment. Perhaps would be better to this as part of the start and have it wait till vault has started and then add the secret.

Reply:
I agree, I don't like it. I'll come up with something different to check whether the vault is up ...

Reply:
Done